### PR TITLE
Enable import order eslint rule and fix related lint errors

### PR DESCRIPTION
**Pull request by rohitkrai03**
**Fixes**: https://issues.redhat.com/browse/ODC-3670

**Analysis / Root cause**: During ESLint upgrade we disabled `import/order` eslint rule due to some issues.

**Solution Description**: Enable `import/order` eslint rule and fix all the lint errors.

**Screen shots / Gifs for design review**: No UI change.

**Unit test coverage report**: No changes in tests. All tests are passing.

Screenshot 2021-06-02 at 9 02 33 PM

**Browser conformance**: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [x] Edge

### CODE REVIEW

File: frontend/packages/kubevirt-plugin/src/components/modals/menu-actions-modals/utils.ts [hunk 1]
```diff
@@ -1,7 +1,6 @@
-import { VMGenericLikeEntityKind } from 'packages/kubevirt-plugin/src/types/vmLike';
-
 import { history } from '@console/internal/components/utils';
 import { getName, getNamespace } from '@console/shared/src/selectors/common';
+import { VMGenericLikeEntityKind } from 'packages/kubevirt-plugin/src/types/vmLike';
```
Comment (andrewballantyne) [inline frontend/packages/kubevirt-plugin/src/components/modals/menu-actions-modals/utils.ts:3]: Hmmm... wonder if this is valid syntax we should allow 🤔

**Review by andrewballantyne — COMMENTED** [review]
Looking good. Few questions inquires to spark conversation.

File: frontend/packages/eslint-plugin-console/lib/config/rules/airbnb-base-overrides.js [hunk 2]
```diff
@@ -34,14 +34,14 @@ module.exports = {
...
-  // ],
-  'import/order': 'off',
+  'import/order': [
+    'error',
+    {
+      groups: ['builtin', 'external', 'internal', 'parent', 'sibling', 'index'],
+      'newlines-between': 'never',
+    },
+  ],
+  // 'import/order': 'off',
```
[the remark this answers is not in the document]
Comment (andrewballantyne) [reply frontend/packages/eslint-plugin-console/lib/config/rules/airbnb-base-overrides.js:44]: @rohitkrai03 This is probably the only requested action by you.

**Comment by andrewballantyne** [comment]
cc @spadgett @TheRealJon FYI - it was disabled in https://github.com/openshift/console/pull/3747

File: frontend/packages/console-app/src/components/cloud-shell/__tests__/useCloudShellAvailable.spec.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { act } from 'react-dom/test-utils';
 import useCloudShellAvailable from '../useCloudShellAvailable';
 import { checkTerminalAvailable } from '../cloud-shell-utils';
+// eslint-disable-next-line import/order
```
[the remark this answers is not in the document]
Comment (rohitkrai03) [reply frontend/packages/console-app/src/components/cloud-shell/__tests__/useCloudShellAvailable.spec.ts:6]: For some reason tests fail if the order of import changes. `useFlag` needs to be imported after `useCloudShellAvailable ` for the `useFlagMock` to work correctly. I've added a comment explaining the reason for disabling the rule.

File: frontend/packages/console-shared/src/utils/utils.ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import { toPath } from 'lodash';
 import { K8sResourceKind } from '@console/internal/module/k8s';
 import { FirehoseResult } from '@console/internal/components/utils/types';
-import { getUID } from '../selectors/common';
 import { JSONSchema6 } from 'json-schema';
```
[the remark this answers is not in the document]
Comment (rohitkrai03) [reply frontend/packages/console-shared/src/utils/utils.ts]: `@console` was considered as an external import so it was mixed up with other npm external imports. Added extra pattern matching for `@console` groups, now these imports will come with `internal` imports but at the top of that group. Also added some logic for `react` to be on top.

File: frontend/packages/dev-console/src/components/import/image-search/ImageStreamNsDropdown.tsx [hunk 5]
```diff
@@ -2,10 +2,10 @@ import * as React from 'react';
 import { useTranslation } from 'react-i18next';
 import { useFormikContext, FormikValues } from 'formik';
 import { ResourceDropdownField } from '@console/shared';
+import * as _ from 'lodash';
```
[the remark this answers is not in the document]
Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/import/image-search/ImageStreamNsDropdown.tsx]: It was not sorting the imports alphabetically within the groups. Added extra options in the rule to sort within the groups as well.

File: frontend/packages/eslint-plugin-console/lib/config/rules/airbnb-base-overrides.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (rohitkrai03) [reply frontend/packages/eslint-plugin-console/lib/config/rules/airbnb-base-overrides.js]: Done.

File: frontend/packages/kubevirt-plugin/src/components/modals/menu-actions-modals/utils.ts (hunk 1 above)
Comment (rohitkrai03) [reply frontend/packages/kubevirt-plugin/src/components/modals/menu-actions-modals/utils.ts:3]: Well, technically its still a valid import but we want to follow the standard practices as much as possible. There might be a lot of instances like this across the repo but for the scope of this PR I didn't want to go into those changes.

Comment (andrewballantyne) [reply frontend/packages/kubevirt-plugin/src/components/modals/menu-actions-modals/utils.ts:3]: Understandable... I saw it and wanted to comment. Wasn't expecting you necessarily to solve this here; just sparking conversation.

You can mark this comment as resolved.

**Review by andrewballantyne — APPROVED** [review]
Going to approve this now as it's growing to be a little unsustainable for GitHub lol. Let's try to merge it asap.

Worth noting, you left the comments above the `import/order` stating why it was disabled.

```
  // TODO This rule was not working as expected prior to eslint-config-airbnb-base v14.0.0
  // Disabling for now since it is causing failures after the update.
  // Sort imports into groups
```

/lgtm

**Comment by rohitkrai03** [comment]
No idea why the bot is going crazy about rebase. The branch completely up to date with upstream master.

**Comment by spadgett** [comment]
@andrewballantyne FYI, the kubevirt tests are broken. Should be fixed by #9140.

**Comment by invincibleJai** [comment]
verified the changes in bot and work as expected

**Comment by rohitkrai03** [comment]
/hold

We are going to wait for the 4.9 master to open as the PR is huge and touches almost all files in console. It would be safer to get this merged next week.

**Comment by rohitkrai03** [comment]
/hold cancel

The PR is ready to be merged now that 4.9 master is open.

**Comment by andrewballantyne** [comment]
@rohitkrai03 Looks like there is another lint --fix you need to run. Sorts are out aligned on a couple files.

**Comment by rohitkrai03** [comment]
> @rohitkrai03 Looks like there is another lint --fix you need to run. Sorts are out aligned on a couple files.

@andrewballantyne Done.

**Pull request merged**